src/Mapping/Driver/AnnotationDriverInterface.php

```diff
@@ -14,15 +15,19 @@
 interface AnnotationDriverInterface extends Driver
 {
     /**
-     * Set annotation reader class
-     * since older doctrine versions do not provide an interface
-     * it must provide these methods:
+     * Set the annotation reader instance
+     *
+     * When originally implemented, `Doctrine\Common\Annotations\Reader` was not available,
+     * therefore this method may accept any object implementing these methods from the interface:
+     *
      *     getClassAnnotations([reflectionClass])
      *     getClassAnnotation([reflectionClass], [name])
      *     getPropertyAnnotations([reflectionProperty])
      *     getPropertyAnnotation([reflectionProperty], [name])
      *
-     * @param object $reader annotation reader class
+     * @param Reader|AttributeReader|object $reader
+     *
+     * @note Providing any object is deprecated, as of 4.0 a `Doctrine\Common\Annotations\Reader` or `Gedmo\Mapping\Driver\AttributeReader` will be required
```

Comment:
```suggestion
     * @param object $reader Providing an object different than `Doctrine\Common\Annotations\Reader` or
     *                       `Gedmo\Mapping\Driver\AttributeReader` is deprecated since version 3.4 and
     *                       will not be allowed in 4.0.
```